modules/danbooru.js

```diff
@@ -0,0 +1,103 @@
+var request = require('request');
+
+/**
+ * The maximum number of attempts if the bot does not get an image at the first try.
+ * This may happen if the first image is marked as Loli/shota. Danbooru will still list them,
+ * but not show them unless you have a gold account. Unfortunately this means we may, once in
+ * a while, get a response without an image path, so we cannot display anything. Therefore we
+ * try again.
+ */
+var maxAttempts = 3;
+
+/**
+ * Request and post an explicit image from danbooru
+ */
+module.exports.nsfw = function(message)
+{
+  requestNsfwImage(message, "explicit");
+}
+
+/**
+ * Request and post a questionable image from danbooru
+ */
+module.exports.ecchi = function(message)
+{
+  requestNsfwImage(message, "questionable");
+}
+
+/**
+ * Request and post a safe image from danbooru
+ */
+module.exports.safe = function(message)
+{
+  requestImage(message, "safe", maxAttempts);
+}
+
+/**
+ * If the channel is marked nsfw request an image.
+ * Otherwise tell the user this is not possible.
+ * @param {*} message The message that was received. Used to determine the channel (and the
+ *                    user, if necessary).
+ * @param {*} rating The rating to use (safe, questionable or explicit. See danbooru)
+ */
+function requestNsfwImage(message, rating)
+{
+  if(message.channel.nsfw)
+  {
+    console.log("Requesting " + rating + " image from danbooru ...");
```

Comment:
Understandable. I ask because I want to maintain enough of a coding standard that everything will be the same once I have a "prettier" go through all the code, also because I know all of you have done the old console.log pattern since I showed it in the hello-bot thing.